app/policies/task_policy.rb

```diff
@@ -0,0 +1,52 @@
+class TaskPolicy < ApplicationPolicy
+
+  authorize :user, allow_nil: true
+
+  def show?
+    project_membership.present?
+  end
+
+  def index?
+    project_member_index.present?
+  end
+
+  def edit?
+    project_membership.present?
+  end
+
+  def new?
+    project_membership.present?
+  end
+
+  def create?
+    project_membership.present?
+  end
+
+  def update?
+    project_membership.present?
+  end
+
+  def destroy?
+    owner?
+  end
+
+
+  private
+
+  def project_membership
+    @project_membership ||= ProjectMembership.find_by(project: record.project, user: user)
+  end
+
+  def project_member_index
+    @project_membership_index ||= ProjectMembership.find_by(project: record.first.project, user: user)
+  end
```

Comment:
It redundant. You can leave first variant with `project_membership`

Reply:
I've created this def so the index def ain't error out, cuz the project_membership for some reason ain't work for it "Wanted to ask you ab it"